_includes/manuals/nightly/3.7_migration.md

```diff
@@ -0,0 +1,185 @@
+
+The first step is to install required dependencies:
+
+    yum install -y foreman-postgresql
+
+Next, stop all your foreman instances and make sure, they stay stopped for
+good. On Red Hat compatible systems, you need to stop the `httpd` and (if
+applicable) `foreman-tasks` service. There is a helper utility called
+`foreman-maintain` when installed to ease the process of shutting down
+services. Make sure that neither Puppet nor a monitoring system automatically
+starts these services again.
+
+You can issue the MySQL command `SHOW FULL PROCESSLIST;` to show you all active
+connections to your MySQL database. Make sure, there are none.
+
+TODO: (WHY DO WE NEED THIS): If you are using the foreman-tasks plugin, you
+need to patch the script we are using to migrate the data. If you are not sure,
+wether you use this plugin, check if you have an entry `Tasks` in Foreman's
+Monitor menu. If you are still not sure, it is safe to apply the patch in any
+case.
+
+```sh
+wget -O /tmp/convert-tasks-patch.diff https://gist.githubusercontent.com/timogoebel/cd29e5206678ca0e0d78222af70583ec/raw/bbe998624098c1c3e656fe2582d33b0d9f9285aa/0001-convert-skip-foreman-tasks-tables.patch
+yum install patch -y
+cd /usr/share/foreman
+
+patch -p1 < /tmp/convert-tasks-patch.diff
+```
+
+The foreman user that has access to your Postgresql database and is used for
+the conversion temporarily needs super user privileges.
+
+```bash
+## postgres@postgres:~  $ su - postgres
+## postgres@postgres:~  $ psql
+psql (9.6.2)
+Type "help" for help.
+
+postgres=# ALTER USER foreman WITH SUPERUSER;
+ALTER ROLE
+postgres=#
+```
+
+#### Migrating
+
+Before proceeding migration, it's a good idea to permanently delete all audit
+records older than few days:
+
+    foreman-rake audits:expire days=10 RAILS_ENV="production"
+
+To permanently delete all reports older than 2 days:
+
+    foreman-rake reports:expire days=2 batch_size=50000 sleep_time=0.000001 RAILS_ENV="production"
+
+The command above deletes records in batches of 50k items, it can take a lot of
+time to process all records. In both cases, progress can be seen in the
+`production.log`.
+
+To migrate the data from one database to the other, we will use a rake task.
+You might already be familiar to this task from upgrading Foreman to a newer
+version.
+
+Command `foreman-rake -T` lists all available tasks. We will be using
+`foreman-rake db:convert:prod2dev` to migrate the data.
+
+Foreman internally uses a development framework called Rails. In Rails, you can
+have a specific configuration for e.g. a development and a production
+environment. Our prod2dev rake task assumes, that the source database is
+configured as the production database and the new target database is configured
+as the development database. All data in the development database will be
+overwritten.
+
+You can configure the new database as Foreman's new development database by
+adding a new connection in `/etc/foreman/database.yml`.
```

Comment:
tbh I would patch the script to read the definitions from elsewhere, we can provide a simple yaml file with source/destination instead of messing up with rails settings.

Reply:
Well, I am not keen doing that, I can test that once or twice on my environment and I don't want this to be rather for the worse.